fix(Space Widget): Add Edit message functionality in space widget & fix double message issue in Space Widget.

pagour98 opened the pull request:
**Problem:**
When the user edits the message from the developer portal API or web client/desktop client the space widget is showing original and as well as edited messages. Also there is no support to show edited message text over edited message moreover when someone edits the message second time the UI shows "You deleted your message" this message should be shown only when the message is deleted.

**Solution:**
Fixed the code in reducer.js and passed the passed the respective props to check where the message is edited or not. Show the message "edited" when the message is edited. 

**Recorded Video: 

https://github.com/user-attachments/assets/1ea0f95b-54ab-4caf-a458-f0916b2f2841

Screenshots:

**Here message has been edited from web client, showing edited text in UI**
<img width="538" alt="Screenshot 2024-09-17 at 13 17 37" src="https://github.com/user-attachments/assets/7841f996-06dd-4925-a8df-2186dfa0682d">

**Here both message edited and original has been deleted from space widget.**
<img width="539" alt="Screenshot 2024-09-17 at 13 17 46" src="https://github.com/user-attachments/assets/bb5b45ae-060f-4a95-83d9-5e6b824f75e4">

**A new message has been sent from space widget.**
<img width="550" alt="Screenshot 2024-09-17 at 13 18 08" src="https://github.com/user-attachments/assets/fa536c5f-4f14-4e5d-aeaf-5817fa43a4e0">

JIRA: SPARK-552328

rsarika commented on packages/node_modules/@webex/redux-module-conversation/src/reducer.js:
```diff
@@ -121,12 +133,17 @@ const processActivities = ({
 
       const threads = ta[parentId] ?? [];
-
+      
```
Try to avoid extra spaces.

pagour98 replied:
Addressed!

rsarika commented on packages/node_modules/@webex/redux-module-conversation/src/reducer.js:
```diff
@@ -121,12 +133,17 @@ const processActivities = ({
 
     if (parentId) {
-      dirtyThreadList.add(parentId);
+        dirtyThreadList.add(parentId);
```
here also

rsarika commented on packages/node_modules/@webex/redux-module-conversation/src/reducer.js:
```diff
@@ -211,6 +228,7 @@ const addActivities = ({state, addedActivities}) => {
 };
 
 export default function reducer(state = initialState, action) {
+
```
here also

pagour98 replied:
**Addressed!**

rsarika commented:
Please attach screenshots of the messages screen

rsarika commented on packages/node_modules/@webex/react-component-activity-item-base/src/index.js:
```diff
@@ -1,8 +1,9 @@
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-
+import {FormattedMessage} from 'react-intl';
```
Is it public component? is it same as web-client?

pagour98 replied:
**Yes, It is same as web-client, we are using FormattedMessage React Component from 'react-intl' library to implement internationalization, we are doing same thing for formatting delete message as well.**

mkesavan13 commented on packages/node_modules/@webex/redux-module-conversation/src/reducer.js:
```diff
@@ -45,6 +45,17 @@ export const initialState = fromJS({
 
+const markEditedActivity = ({activity, activities}) =>
+  activities.map((activityItem) => {
+    if (activityItem.id === activity.parent.id && activity.object.displayName) {
+      activityItem.object.displayName = activity.object.displayName;
```
Are we overriding the `displayName` here or the check is incorrect?

I see that we check if the `activity.object.displayName` exists and override it. Should we be rather checking if it doesn't exist?

mkesavan13 replied:
Ah. I just read the condition correctly and there's `activityItem` and `activity` that I missed to notice. Resolving

mkesavan13 commented on packages/node_modules/@webex/redux-module-conversation/src/reducer.js:
```diff
@@ -45,6 +45,17 @@ export const initialState = fromJS({
+  activities.map((activityItem) => {
+    if (activityItem.id === activity.parent.id && activity.object.displayName) {
+      activityItem.object.displayName = activity.object.displayName;
+      return Object.assign(activityItem, activity.object.displayName, {
```
If we have already assigned the displayName in previous line, why are we assigning it again and returning?

pagour98 replied:
@mkesavan13 Good catch! 
I have updated the code.

mkesavan13 commented on packages/node_modules/@webex/react-component-activity-item-base/src/index.test.js:
```diff
@@ -108,4 +109,28 @@ describe('ActivityItemBase component', () => {
+    expect(component).toMatchSnapshot();
+  });
+
+  it('renders properly when message is not edited', () => {
+    props.isMessageEdited = false;
+    renderer.render(
+      <ActivityItemBase>
+        {child}
+      </ActivityItemBase>
+    );
+    const component = renderer.getRenderOutput();
+    
+    expect(component).toMatchSnapshot();
+  });
```
Shouldn't we already have a test case for this? A regular test case? We shall cover this as part of that case instead.

pagour98 replied:
@mkesavan13, We do not have edit message feature supported yet in space widget. So we have added a edit message functionality when the message is edited from Web/Desktop Client. So, there is no test case related to edit.

mkesavan13 replied:
I mean, 'renders properly when message is not edited' is the case, that is like the default message that comes into the widget right?

mkesavan13 reviewed (commented):
Changes mostly seem good. A couple of comments and please remove all the extra spaces as mentioned by Ravi, regenerate the snapshots.

Finally, please change the PR title as per semantic-release standards with proper commit type, component and message. 

https://semantic-release.gitbook.io/semantic-release#commit-message-format

pagour98 commented:
> Please attach screenshots of the messages screen

@rsarika, SS has been added!

pagour98 commented:
> Changes mostly seem good.
> 
> * A couple of comments below
> * Please remove all the extra spaces as mentioned by Ravi, regenerate the snapshots.
> * Please change the PR title as per semantic-release standards with proper commit type, component and message.
> * Add the Vidcast to PR description itself for easy access
> 
> https://semantic-release.gitbook.io/semantic-release#commit-message-format

@mkesavan13 **I have addressed all these comments.**

pagour98 commented on packages/node_modules/@webex/react-container-activity-list/src/__snapshots__/index.test.js.snap:
```diff
@@ -54,6 +54,7 @@ exports[`ActivityList renders properly 1`] = `
             className="webex-published published"
           >
             1/3/15, 5:22 PM
+             
```
**This is getting added automatically whenever we push changes.**

pagour98 commented on packages/node_modules/@webex/react-container-activity-list/src/__snapshots__/index.test.js.snap:
```diff
@@ -217,6 +218,7 @@ exports[`ActivityList renders properly 1`] = `
             className="webex-published published"
           >
             1/6/15, 9:17 PM
+             
```
**Jest is adding up these lines automatically to the snapshots.**

pagour98 commented on packages/node_modules/@webex/react-component-activity-item-base/src/__snapshots__/index.test.js.snap:
```diff
@@ -30,6 +30,7 @@ exports[`ActivityItemBase component renders properly 1`] = `
         className="webex-published published"
       >
         2016-09-20T19:52:57.186Z
+         
```
**These below new lines are getting automatically by jest whenever we run test case.**

mkesavan13 reviewed (approved):
Changes looks good to me. Approving.

For the spaces that jest added in the snapshots, I think it is a side effect of a blank line added in the component. See if you can find that unwanted blank line and remove, these should then go away from the snapshot.
